images/cli/Dockerfile.rhel

```diff
@@ -1,10 +1,12 @@
 FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.10 AS builder
 WORKDIR /go/src/github.com/openshift/origin
 COPY . .
-RUN make build WHAT=cmd/oc
+RUN make build WHAT=cmd/oc; \
+    mkdir -p /tmp/build; \
+    cp /go/src/github.com/openshift/origin/_output/local/bin/linux/$(go env GOARCH)/oc /tmp/build/oc
```

Comment:
Why this extra step to copy to /tmp/build/oc in this stage rather than just copying from the arch specific location in the next one?

Reply:
Golang doesn't always exist in the base container, so we would have to use $(uname -m) to determine the correct architecture.

`Copy --from= builder /path/$(uname -m) `
 Unfortunately, the output of uname doesn't always equal  $(go env GOARCH)
ie.  uname could return x86_64 not amd64.

The other option is to have hack/build-go.sh output to a non-arch specific directory.
https://github.com/openshift/origin/blob/master/hack/lib/build/binaries.sh#L259
I think changing that logic could break cross-compiling windows binaries and other things I don't know about. 

cc: @sosiouxme

Reply:
Ok, that makes sense.